lib/formatters/default.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const path = require('path');
+
+class DefaultPrinter {
+  constructor(options = {}) {
+    this.delegates = [];
+    this.console = options.console || console;
+
+    if (options.json) {
+      options.format = 'json';
+    }
```

Comment:
I would actually like to differentiate between the `--json` flag and the internal formatters. Specifically, I'd like to diverge the formats and have a new JSON formatter that writes files to disk (instead of emitting stdout).

Reply:
Fixed.